### PR TITLE
compilers/detect: remove unsupported -cpp flag for Clang preprocessor detection

### PR DESCRIPTION
**Pull request by tambry**
Clang's resource files, e.g. /usr/share/clang/clang++.cfg, can be used to bump the default standard level across the system.  
However due to llvm/llvm-project#61641 `clang++ -std=c++17 -E -dM -` doesn't work. The workaround is to pass the language explicitly.  
4ad792e158b6059eb847dd0562aff9bd7029981f fixed the issue by passing the language explicitly, but started passing the `-cpp` flag, which Clang doesn't support.  
Basically Clang would always fallback to the second detection attempt as a result. Remove the unsupported flag and the above scenarios works now too. 🙂

See-also: https://github.com/llvm/llvm-project/issues/61641  
Fixes: 4ad792e158b6059eb847dd0562aff9bd7029981f

### CODE REVIEW

File: mesonbuild/compilers/detect.py [hunk 1]
```diff
@@ -1440,7 +1440,7 @@ def _try_obtain_compiler_defines(args: T.List[str]) -> str:
         # so first, try with '-cpp -x lang', then fallback without given it's less
         # portable. We try with '-cpp' as GCC needs it for Fortran at least, and
```
Comment (thesamesam) [inline mesonbuild/compilers/detect.py:1441]: Please update the comment.
